back/app/services/side_fx_vote_service.rb

```diff
@@ -9,27 +9,33 @@ def after_create(vote, current_user)
     if vote.votable_type == 'Initiative'
       AutomatedTransitionJob.perform_now
     end
-    type = votable_type(vote)
-    LogActivityJob.perform_later(vote, "#{type}_#{vote.mode}voted", current_user, vote.created_at.to_i)
+
+    action = "#{votable_type(vote)}_#{vote.mode}voted"
+    log_activity_job(vote, action, current_user)
   end
 
   def before_destroy(vote, current_user); end
 
-  def after_destroy(frozen_vote, current_user)
-    serialized_vote = clean_time_attributes(frozen_vote.attributes)
-    type = votable_type(frozen_vote)
-    LogActivityJob.perform_later(
-      encode_frozen_resource(frozen_vote),
-      "canceled_#{type}_#{frozen_vote.mode}vote",
-      current_user,
-      Time.now.to_i,
-      payload: { vote: serialized_vote }
-    )
+  def after_destroy(vote, current_user)
+    action = "canceled_#{votable_type(vote)}_#{vote.mode}vote"
+    log_activity_job(vote, action, current_user)
   end
 
   private
 
   def votable_type(vote)
     vote.votable_type.underscore
   end
+
+  def log_activity_job(vote, action, current_user)
+    serialized_vote = clean_time_attributes(vote.attributes)
+
+    LogActivityJob.perform_later(
+      encode_frozen_resource(vote),
+      action,
+      current_user,
+      Time.now.to_i,
+      payload: { vote: serialized_vote }
+    )
+  end
```

Comment:
This pattern is used in multiple side_fx services (more than 20). Would it be a good idea to move this to something more central? For example; [side_fx_helper.rb](https://github.com/CitizenLabDotCo/citizenlab/blob/master/back/app/services/concerns/side_fx_helper.rb
). This could reduce code duplication and simplify many side_fx services. It could be a nice little 10% project.

Thoughts?

Reply:
Yes, definitely a good idea!

Reply:
Ticket created: https://citizenlab.atlassian.net/browse/CL-2158